src/app/lib/providers/trakttv.js

```diff
@@ -1,676 +1,241 @@
-                'trakt-api-version': '2',
-                'trakt-api-key': CLIENT_ID
-            },
-            body: JSON.stringify(postVariables)
-        }, function (error, response, body) {
-            if (error || !body) {
-                defer.reject(error);
-            } else if (response.statusCode >= 400) {
-                defer.resolve({});
-            } else {
-                defer.resolve(Common.sanitize(JSON.parse(body)));
-            }
-        });
-
-        return defer.promise;
-    };
-
-    TraktTv.prototype.delete = function (endpoint, getVariables) {
-        var defer = Q.defer();
-
-        getVariables = getVariables || {};
-
-
-        var requestUri = API_ENDPOINT.clone()
-            .segment(endpoint)
-            .addQuery(getVariables);
-
-        request({
-            method: 'DELETE',
-            url: requestUri.toString(),
-            headers: {
-                'Content-Type': 'application/json',
-                'Authorization': 'Bearer ' + Settings.traktToken,
-                'trakt-api-version': '2',
-                'trakt-api-key': CLIENT_ID
-            }
-        }, function (error, response, body) {
-            if (error) {
-                defer.reject(error);
-            } else {
-                defer.resolve({});
+        this.client = new Trakt({
+            client_id: Settings.trakttv.client_id,
+            client_secret: Settings.trakttv.client_secret,
+            plugins: ['ondeck', 'matcher', 'images'],
+            options: {
+                images: {
+                    smallerImages: true,
+                    fanartApiKey: Settings.fanart.api_key,
+                    tvdbApiKey: Settings.tvdb.api_key,
+                    tmdbApiKey: Settings.tmdb.api_key
+                }
             }
         });
 
+        this.authenticated = false;
 
-        return defer.promise;
-    };
-
-    TraktTv.prototype.calendars = {
-        myShows: function (startDate, days) {
-            var endpoint = 'calendars/my/shows';
-
-            if (startDate && days) {
-                endpoint += '/' + startDate + '/' + days;
-            }
-
-            return this.get(endpoint)
-                .then(function (item) {
-                    var calendar = [];
-                    for (var i in item) {
-                        calendar.push({
-                            show_title: item[i].show.title,
-                            show_id: item[i].show.ids.imdb,
-                            aired: item[i].first_aired.split('T')[0],
-                            episode_title: item[i].episode.title,
-                            season: item[i].episode.season,
-                            episode: item[i].episode.number,
-                        });
-                    }
-                    return calendar;
-                });
-        }
-    };
+        // auto sign-in when database is loaded
+        App.vent.on('db:ready', this.reAuthenticate.bind(this));
```

Comment:
is it any useful to have any trakt before db is ready ? shouldn't we make this a promise and just chain trakt (and everything that depends on that) on the loading of that promise ?